packages/cli/src/modules/insights/insights-pruning.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Service } from '@n8n/di';
+import { Logger } from 'n8n-core';
+
+import { Time } from '@/constants';
+
+import { InsightsByPeriodRepository } from './database/repositories/insights-by-period.repository';
+import { InsightsConfig } from './insights.config';
+
+@Service()
+export class InsightsPruningService {
+	private pruneInsightsTimeout: NodeJS.Timeout | undefined;
+
+	private isStopped = false;
+
+	private readonly delayOnError = Time.seconds.toMilliseconds;
+
+	constructor(
+		private readonly insightsByPeriodRepository: InsightsByPeriodRepository,
+		private readonly config: InsightsConfig,
+		private readonly logger: Logger,
+	) {
+		this.logger = this.logger.scoped('insights');
+	}
+
+	get isPruningEnabled() {
+		return this.config.maxAgeDays > -1;
+	}
```

Comment:
What I meant here was, we could have the check at callsite:

```ts
	startTimers() {
		this.compactionService.startCompactionTimer();
		this.collectionService.startFlushingTimer();
		if (isPruningEnabled) this.pruningService.startPruningTimer();
		this.logger.debug('Started compaction, flushing and pruning schedulers');
	}
```

And codify the expectation:

```ts
	startPruningTimer() {
		strict(this.isStopped);
		// ...
}
```

Unless this wouldn't work for some reason?